testsuite/cluster-test/src/tx_emitter.rs

```diff
@@ -508,13 +508,30 @@ impl TxEmitter {
         job.stats.accumulate()
     }
 
+    pub async fn periodic_stat(&mut self, job: &EmitJob, duration: Duration, interval_secs: u64) {
+        let deadline = Instant::now() + duration;
+        let mut prev_stats: Option<TxStats> = None;
+        while Instant::now() < deadline {
+            let window = Duration::from_secs(interval_secs);
+            tokio::time::delay_for(window).await;
+            let stats = self.peek_job_stats(job);
+            let delta = &stats - &prev_stats.unwrap_or_default();
+            prev_stats = Some(stats);
+            println!("{}", delta.rate(window));
+        }
+    }
+
     pub async fn emit_txn_for(
         &mut self,
         duration: Duration,
         emit_job_request: EmitJobRequest,
+        periodic_stat: bool,
```

Comment:
It looks like pretty much all usages of `emit_txn_for` do not need periodic_stat.
Do you think you can just introduce new function `emit_txn_for_with_stats` and keep places that don't need stats as is?

Also note that cluster_test/main.rs:330-342. It seems like we can reuse this piece of code and have it as `emit_txn_for_with_stats`?

Reply:
hmmmm, nice catch.... actually I did try to make another similar emit_txn fn with stats at first but I got one rust error cannot solve if I make if condition with two emit_txns. :) I can try to solve that.

```
Also note that cluster_test/main.rs:330-342. It seems like we can reuse this piece of code and have it as emit_txn_for_with_stats?
```
this is good found, we can reuse more code there